graphql/mutation_resolver.go

```diff
@@ -754,6 +754,19 @@ func (r *mutationResolver) SpawnHost(ctx context.Context, spawnHostInput *SpawnH
 		return nil, err
 	}
 
+	if !usr.HasDistroCreatePermission() {
```

Comment:
Could we add a similar comment here to the one I mentioned below (or deduplicate the business logic somehow? I don't think we have a common API that a facade could return and the graphql/rest api could interpret and send back the appropriate response)